compiler/jsgen.nim

```diff
@@ -456,7 +467,10 @@ proc arith(p: PProc, n: PNode, r: var TCompRes, op: TMagic) =
     gen(p, n.sons[1], x)
     gen(p, n.sons[2], y)
     let trimmer = unsignedTrimmer(n[1].typ.skipTypes(abstractRange).size)
-    r.res = "(($1 $2) >>> $3)" % [x.rdLoc, trimmer, y.rdLoc]
+    if p.target == targetPHP:
+      r.res = "(($1 $2) >= 0) ? (($1 $2) >> $3) : ((($1 $2) & 0x7fffffff) >> $3) | (0x40000000 >> ($3 - 1))" % [x.rdLoc, trimmer, y.rdLoc]
```

Comment:
This cannot work since $2 is evaluated multiple times. If $2 has side-effects this breaks. There might be other cases in the JS backend where this can happen. We need to introduce ``genOnce(p, n.sons[1], x)`` that stores the location in a temporary if it's a complex expression. But then the complex vs simple control flow analysis needs to know about this as well unless genOnce() ensures it stays an expression. I need to do this, just mark it with 'XXX prevent multi evaluations'.